url49584c1d41_Arrem_GGS_tgz-pJ8-com_gamezgalaxy_test_console_MainJ8/src/com/gamezgalaxy/GGS/world/convert/DatToGGS.java

```diff
@@ -0,0 +1,181 @@
+/*******************************************************************************
+ * Copyright (c) 2012 GamezGalaxy.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the GNU Public License v3.0
+ * which accompanies this distribution, and is available at
+ * http://www.gnu.org/licenses/gpl.html
+ ******************************************************************************/
+package com.gamezgalaxy.GGS.world.convert;
+
+import java.io.*;
+import java.text.DateFormat;
+import java.util.zip.GZIPOutputStream;
+import java.util.zip.GZIPInputStream;
+
+/**
+ * This class was taken from the Minecraft Wiki
+ * and was slightly modified
+ *
+ */
+public class DatToGGS {
+	public com.mojang.minecraft.level.Level level = null;
+	private int width;
+	private int height;
+
+	public DatToGGS() {
+	}
+
+	public void load(String filename) {
+		FileInputStream fis = null;
+		GZIPInputStream gzis = null;
+		ObjectInputStream in = null;
+		DataInputStream inputstream = null;
+		try {
+			fis = new FileInputStream(filename);
+			gzis = new GZIPInputStream(fis);
+			inputstream = new DataInputStream(gzis);
+			if((inputstream.readInt()) != 0x271bb788) {
+				fis.close();
+				gzis.close();
+				inputstream.close();
+				return;
+			}
+			if((inputstream.readByte()) > 2) {
+				System.out.println("Error: Level version > 2, this is unexpected!");
+				fis.close();
+				gzis.close();
+				inputstream.close();
+				return;
+			}
+			in = new ObjectInputStream(gzis);
+			level = (com.mojang.minecraft.level.Level)in.readObject();
+			inputstream.close();
+			in.close();
+			System.out.println("Loading level "+filename+" successful");
+			fis.close();
+			gzis.close();
+		} catch(IOException ex) {
+			ex.printStackTrace();
+		} catch(ClassNotFoundException ex) {
+			ex.printStackTrace();
+		}
+		level.initTransient();
+		width = level.width;
+		height = level.height;
+		}
+
+	// save in file called filename
+	public void save(String filename) {
+		FileOutputStream fos = null;
+		GZIPOutputStream gzos = null;
+		ObjectOutputStream out = null;
+		DataOutputStream outputstream = null;
+		try {
+			fos = new FileOutputStream(filename);
+			gzos = new GZIPOutputStream(fos);
+			outputstream = new DataOutputStream(gzos);
+			outputstream.writeInt(0x271bb788);
+			outputstream.writeByte(2);
+			out = new ObjectOutputStream(gzos);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *out*. The resource is referenced at lines 75, 76, 77, 78, 79, 80. The resource is closed at lines 81, 82. There are other execution paths that do not contain closure statements, for example, when ObjectOutputStream.writeObject throws an exception. Ensure *out* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).